### PR TITLE
WebUI: fix WebUI unit tests

### PR DESCRIPTION
**Pull request by pvomacka**
This pull request fixes WebUI unit tests.

And there are also commits from PR #628 which should be removed once the PR #628 will be pushed. But they are necessary for this PR.

Information for reviewer:
There are several steps which needs to be done to run the tests.

1. run configure in root directory of the project (.makerpms.sh also works well)
2. run $ make in install/ui/src/libs
3. tests needs to be opened in Firefox (Chrome has issues with XMLHttpRequest to local files)

https://pagure.io/freeipa/issue/6974

### CODE REVIEW

File: install/ui/test/ipa_tests.js [hunk 1]
```diff
@@ -212,8 +220,14 @@ test("Testing unsuccessful rpc.command().", function() {
 
         equals(data.method, object+'_'+method, "Checking method");
 
+        // By default all rpc calls contain version of API
+        $.extend(options, { 'version': window.ipa_loader.api_version});
+
         same(data.params, [args, options], "Checking parameters");
 
+        // remove api version from options object
+        delete options.version;
```
Comment (pvoborni) [inline install/ui/test/ipa_tests.js]: Comment describes the obvious - what it does. But it should describe why it does it which is not clear.

File: Makefile.am [hunk 2]
```diff
@@ -245,10 +245,6 @@ jslint-ui: $(top_builddir)/install/ui/src/libs/loader.js
 	rm js/libs;						\
 	rmdir js
 
-jslint-ui-test:
```
Comment (pvoborni) [inline Makefile.am]: jslint-ui-tests shouldn't be removed. Some of the tests remains there so it should stay and also should be kept in jslint target.

File: install/ui/test/ipa_tests.js (hunk 1 above)
Comment (pvomacka) [inline install/ui/test/ipa_tests.js]: Deleting the version is not necessary, so at the end I remove this line.

**Comment by pvomacka** [comment]
@pvoborni Thank you for the review. I updated README and also updated our wiki page with information about how to run FreeIPA unit tests.

**Comment by pvoborni** [comment]
But one nitpick in REAME file:

3. Go to install/ui/libs/ directory

The folder is wrong, it should be install/ui/src/libs

Also for future it would be good to run it e.g. in https://github.com/qunitjs/node-qunit or other headless runner. And then executing it in travis or PR-CI.

**Comment by pvomacka** [comment]
Hi @pvoborni, thanks for review. 
I'll fix the path in README file. 

For future I'm planning to run those tests in our Travis CI (can be done also in PR-CI) using Grunt. I will also check node-qunit whether it fits our need better or not.

**Comment by pvomacka** [comment]
I updated the README and rebased patches on current master.

**Comment by pvomacka** [comment]
master:

* 46fba2128fb42c5c2cbea60ff288f3180d729042 Add loader.js into requirements of all HTML unit test files
* 9e0db0759abe35d786f6570234afb04d353ae6df Add up to date JSON files
* 4fb52371f27f4013a8e30c09ccaed75da5a08404 Fixes for ipa_test.js
* 11e19cd5d5e475350d6de29d9cb2920c3ab99110 Fixes for entity_tests.js
* 63f7575cb38982575342a6856b7d322c14fc83f6 Fixes for aci_tests.js
* bb5582d52bc7ce80e6cfc691cc4bb10a9beb307b Fixes for widget_tests.js
* 6b70b91de4d241b999d62cc56051c8af364d3630 Fixes details_test.js
* 8c2dbece59b9ff1ee8d247e940e2e776a660fec8 WebUI: Update unit test README

**Pull request closed without merging**